core/src/com/unciv/logic/BackwardCompatibility.kt

```diff
@@ -144,6 +144,14 @@ object BackwardCompatibility {
                     unit.promotions.addPromotion(startingPromo, true)
     }
 
+    /** Upgrade the uniques from deprecated format to the new more general one **/
+    fun GameInfo.updateGreatGeneralUniques() {
+        ruleSet.units.values.filter { it.uniques.contains("Bonus for units in 2 tile radius 15%") }.forEach {
+            it.uniques.remove("Bonus for units in 2 tile radius 15%")
+            it.uniques.add("[+15]% Strength bonus for [Military] units in [2] tiles")
+        }
+    }
+
```

Comment:
I'd personally put a `// Deprecated since 4.1.1` above this function so we don't forget to remove it when deprecating uniques (searching for `deprecated` would then reveal it).
Alternatively, doing `it.uniques.any { it.isOfType(UniqueType.GreatGeneralDeprecated)) } ` instead of `it.uniques.contains("Bonus for units in 2 tile radius 15%) }` would also make the connection between this function and the deprecated unique more clear

Reply:
The current comment says "Upgrade the uniques from **deprecated** format to the new more general one", so the search should find it.

Reply:
Or @ Deprecate the fun itself. Also has precedents if I remember correctly. But don't @ Deprecate `Stat.happiness`, the respect-lacking bigfarts are already trying that one and will hopefully fail soon.